### DISCUSSION TITLE
SCSI ID hardware switch support.

### DISCUSSION THREAD

**Discussion by matttechwood** [ZuluSCSI/ZuluSCSI-firmware#353] [Ideas]
Unless I missed something, is there a way to use ZuluSCSI RP 2040 (rev2023a) with a device that has a hardware ID switch?   This old telecom server has 4 wires that go to the original drive that jumps and Un jumps ID pins, to select what SCSI ID you want to make your drive. the device is like an old car odometer but there are two buttons to go up and down, it has a selection of 0-6. by chance Is this what the location of J307 is for?

thank you so much, 

-Matt

**Comment by aperezbios** [comment]
@matttechwood no, there's no support for those sorts of hardware SCSI ID switches. J307 is for an i2c interface, although it can be abused for other purposes. That said, you would need a secondary microcontroller which can decode the switch state, and the SCSI ID selector would need to be attached to that, which could then communicate this sort of information to the RP2040 via i2c. It would also potentially be possible to accomplish using an i2c GPIO expander. 

That said, since the ZuluSCSI can emulate multiple SCSI devices at the same time, you run in to a situation where you would either need to lose that functionality, or create a way at the firmware level map the physical switch to a specific device. In short, it's a lot easier said than done. You're not the first person to ask for this sort of functionality, and I'm sure you won't be the last :)
